templates/kotsadm-statefulset.yaml

```diff
@@ -28,8 +28,15 @@ spec:
         - name: SHARED_PASSWORD_BCRYPT
           valueFrom:
             secretKeyRef:
+{{- if .Values.passwordSecretRef }}
+{{- with .Values.passwordSecretRef }}
+              key: {{ .key }}
+              name: {{ .name }}
+{{- end }}
+{{- else }}
```

Comment:
I believe we will also need to somehow tell KOTS what this new secret name is as it updates it at runtime when the password is changed. The `kotsadm-password` name is currently hard-coded in a few places.

One example where it updates the secret: https://github.com/replicatedhq/kots/blob/v1.109.0/pkg/password/password.go#L80